pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod.go

```diff
@@ -55,19 +55,28 @@ func ExtractPod(p *corev1.Pod) *model.Pod {
 	podModel.Status = computeStatus(p)
 	podModel.ConditionMessage = getConditionMessage(p)
 
-	for _, c := range p.Spec.Containers {
+	podModel.ResourceRequirements = GetModelResourceRequirements(p.Spec.Containers, p.Spec.InitContainers)
+
+	return &podModel
+}
+
+func GetModelResourceRequirements(containers []corev1.Container, initContainers []corev1.Container) []*model.ResourceRequirements {
+	resReq := make([]*model.ResourceRequirements, 0)
+	for _, c := range containers {
 		if modelReq := convertResourceRequirements(c.Resources, c.Name, model.ResourceRequirementsType_container); modelReq != nil {
-			podModel.ResourceRequirements = append(podModel.ResourceRequirements, modelReq)
+			resReq = append(resReq, modelReq)
 		}
 	}
 
-	for _, c := range p.Spec.InitContainers {
+	for _, c := range initContainers {
 		if modelReq := convertResourceRequirements(c.Resources, c.Name, model.ResourceRequirementsType_initContainer); modelReq != nil {
-			podModel.ResourceRequirements = append(podModel.ResourceRequirements, modelReq)
+			resReq = append(resReq, modelReq)
 		}
 	}
-
-	return &podModel
+	if len(resReq) == 0 {
+		return nil
+	}
```

Comment:
why do we need this?

Reply:
This is a small hack, because currently the unit tests don't have that field set, meaning they expect the default value, in this case `nil` i.e.
```
model.Deployment.ResourceRequirements = nil
```

So returning explicitly nil was a choice out of laziness, I remembered this pattern being used somewhere else in the code base + I didn't see too many cons/unexpected behaviours in doing so.

Do you prefer to return the empty slice and edit the unit tests?